irohad/consensus/yac/impl/yac_crypto_provider_impl.cpp

```diff
@@ -41,26 +41,37 @@ namespace iroha {
       }
 
       bool CryptoProviderImpl::verify(VoteMessage msg) {
-        return iroha::verify(
-            iroha::sha3_256(
-                PbConverters::serializeVote(msg).hash().SerializeAsString())
-                .to_string(),
-            msg.signature.pubkey,
-            msg.signature.signature);
+        auto serialized =
+            PbConverters::serializeVote(msg).hash().SerializeAsString();
+        auto blob = shared_model::crypto::Blob(serialized);
+
+        return shared_model::crypto::CryptoVerifier<>::verify(
+            msg.signature->signedData(), blob, msg.signature->publicKey());
       }
 
       VoteMessage CryptoProviderImpl::getVote(YacHash hash) {
+        logger::Logger log = logger::log("YacCryptoProvider::getVote");
```

Comment:
Maybe localize it and place to the building error? So in a good-case scenario it won't be created at all

Reply:
good point